generic/Storage.c

```diff
@@ -39,6 +39,27 @@ static int torch_Storage_(new)(lua_State *L)
       lua_pop(L, 1);
     }
   }
+  else if(lua_type(L, index) == LUA_TUSERDATA)
+  {
+    THStorage *src = luaT_checkudata(L, index, torch_Storage);
+    if (src->flag & TH_STORAGE_VIEW) {
+      // don't allow chained views
+      src = src->view;
```

Comment:
Oh I see. Yep, good point, not super intuitive. So we can still keep that code, but set the ptr to the original src. This way we still don't have to do chained references, but the pointer will use the chained offset. It would look like:

```
real *ptr = src->data;
if (src->flag & TH_STORAGE_VIEW) {
  src = src->view;
}
```

Essentially just moving the ptr set above. Would that be ok?